### PR TITLE
Migrate core library to the new Stdlib API/projects system

### PR DESCRIPTION
**Pull request by sezna**
This is a follow-up to #1912, and includes an associated RCA bug fix (#1918 ) that was not exposed until I tried to re-export something from `Core`.

This also makes a subtle change to the compiler where the prelude is no longer created by default in the namespace tree, and is rather created when we compile `core.qs`. This was needed because we no longer allow "clobbering namespaces", that is, re-opening a namespace from an _external_ package.

### CODE REVIEW

File: compiler/qsc_frontend/src/resolve.rs [hunk 1]
```diff
@@ -182,6 +182,12 @@ pub(super) enum Error {
         #[label]
         span: Span,
     },
+    #[error("this namespace clobbers (replaces) an existing external namespace of the same name")]
```
Comment (swernli) [inline compiler/qsc_frontend/src/resolve.rs:185]: nitpick: I know "clobbered" is the right and even common term of art for this, but it feels odd in an error message for some reason. Perhaps "overwrites" would be similarly descriptive?

File: compiler/qsc_frontend/src/resolve/tests.rs [hunk 2]
```diff
@@ -204,11 +204,17 @@ fn compile(
...
+
     let mut globals = super::GlobalTable::new();
     let mut errors = globals.add_local_package(&mut assigner, &package);
     let mut resolver = Resolver::new(globals, dropped_names);
+    // no errors here
     resolver.bind_and_resolve_imports_and_exports(&package);
+    // what about here?
+    println!("i have {} errors", resolver.errors.len());
     resolver.with(&mut assigner).visit_package(&package);
+    println!("now i have {} errors", resolver.errors.len());
```
Comment (swernli) [inline compiler/qsc_frontend/src/resolve/tests.rs:217]: Looks like these might have been leftover from testing.

**Review by swernli — APPROVED** [review]
A few minor things to fix, but otherwise this is looking good!

File: compiler/qsc_frontend/src/resolve/tests.rs (hunk 2 above)
Comment (sezna) [reply compiler/qsc_frontend/src/resolve/tests.rs]: oops 😓 fixed!

**Review by cesarzc — APPROVED** [review]

**Pull request merged**